src/main/java/org/reactome/release/qa/check/CoV2InfectionPathwayEventCheck.java

```diff
@@ -0,0 +1,176 @@
+            if (issues.size() > 0) {
+                report.addLine(getReportLine(cov2Event, String.join("|",issues)));
+            }
+        }
+        report.setColumnHeaders(getColumnHeaders());
+        return report;
+    }
+
+    /**
+     * Checks if instance created dateTime is before most recent modified instance's dateTime. Also checks if
+     * any modified instance is present -- it should be given that there was at least 1 required modification from the
+     * CoV-1-to-CoV-2 merge.
+     * @param cov2Event - GKInstance, Event contained within 'SARS-CoV-2 Infection' Pathway.
+     * @return - boolean, true if modification property looks correct, false if not.
+     * @throws Exception, thrown by MySQLAdaptor.
+     */
+    private boolean hasRecentlyModifiedSummation(GKInstance cov2Event) throws Exception {
+       for (GKInstance summation : (Collection<GKInstance>) cov2Event.getAttributeValuesList(ReactomeJavaConstants.summation)) {
+            GKInstance createdInst = (GKInstance) summation.getAttributeValue(ReactomeJavaConstants.created);
+            List<GKInstance> modifieds = summation.getAttributeValuesList(ReactomeJavaConstants.modified);
+            if (modifieds.size() > 0) {
+                // At time of writing (August 2020), nothing is reported from this clause, but it is a good check.
+                GKInstance mostRecentModifiedInst = modifieds.get(modifieds.size() - 1);
+                // Only check Year/Month/Day since the changes took place more than a day before Curators got to look at them in gk_central.
+                int createdDateTime = Integer.valueOf(createdInst.getAttributeValue(ReactomeJavaConstants.dateTime).toString().split(" ")[0].replaceAll("-", ""));
+                int modifiedDateTime = Integer.valueOf(mostRecentModifiedInst.getAttributeValue(ReactomeJavaConstants.dateTime).toString().split(" ")[0].replaceAll("-", ""));
+                if (createdDateTime > modifiedDateTime) {
+                    return false;
+                }
+            // Should be at least 1 modified instance in CoV-1-to-CoV-2 instances.
+            } else {
+                return false;
+            }
+       }
+        return true;
+    }
+
+    /**
+     * Checks if Event has a 2020 literatureReference. This is because CoV-2 evidence was very recent when this was being evaluated.
+     * @param cov2Event - GKInstance, Event contained within 'SARS-CoV-2 Infection' Pathway.
+     * @return - boolean, true if at least 1 2020 literatureReference, false if not.
+     * @throws Exception, thrown by MySQLAdaptor.
+     */
+    private boolean hasRecentLiteratureReference(GKInstance cov2Event) throws Exception {
+
+        boolean has2020LiteratureReference = false;
+        // Iterature through all literatureReference instances, checking the 'year' attribute.
+        for (GKInstance literatureReference : (Collection<GKInstance>) cov2Event.getAttributeValuesList(ReactomeJavaConstants.literatureReference)) {
```

Comment:
This could be rewritten as a while loop that breaks as soon as it finds a literature reference from 2020, as there's no need to examine ALL literature references if you only care about one being from 2020.

Reply:
Fair point

Reply:
I know I'm in the minority on this, but you could also consider streaming the literature references and using the method `anyMatch` to check for the condition you're looking for.